app/controllers/discovered_hosts_controller.rb

```diff
@@ -25,8 +26,19 @@ def index
 
   def show
     # filter graph time range
+    @general =    {:Highlights => ["size", "productname", "memorysize", "interface", "manufacturer", "architecture", "macaddress", "processorcount", "physicalprocessorcount", "discovery_boot"]}
+    @highlights = {:Storage => ["blockdevice"],
+                   :Hardware => ["hardw", "manufacturer", "memo", "process"],
+                   :Network => ["ipaddress", "interfaces", "dhcp", "fqdn", "hostname", "link", "mtu", "net"] + @host.facts_hash["interfaces"].split(","),
+                   :Software => ["bios", "os", "discovery"]}
+    @facts = @host.facts_hash
+    @facts.delete("memorysize_mb")
+    @facts.each do |key, val|
+       @highlights.merge!({:IPMI => ["ipmi"]}) if key.to_s.downcase.include?("ipmi")
+       @facts[key] = number_to_human_size(val)  if key.to_s.downcase.include?("blockdevice") && key.to_s.downcase.include?("_size")
```

Comment:
Can you use (case insensitive) regular expressions which are faster than this? Also you can leverage begin/end of the string.